### PR TITLE
Implement pre-commit

### PR DESCRIPTION
**Pull request by BobDotCom**
## Summary

Implements pre-commit hooks

## Information

- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Review by Lulalaby — APPROVED** [review]
if this breaks, not my fault

File: discord/embeds.py [hunk 1]
```diff
@@ -405,7 +395,7 @@ def colour(self) -> MaybeEmpty[Colour]:
         return getattr(self, "_colour", EmptyEmbed)
 
     @colour.setter
-    def colour(self, value: Union[int, Colour, _EmptyEmbed]):  # type: ignore
+    def colour(self, value: int | Colour | _EmptyEmbed):  # type: ignore
```
Comment (Dorukyum) [inline discord/embeds.py:398]: We're changing the base python version to 3.10?

Comment (Lulalaby) [reply discord/embeds.py:398]: See https://discord.com/channels/881207955029110855/929023758742749244/1026859802808307742

Comment (BobDotCom) [reply discord/embeds.py:398]: TL;DR: `from __future__ import annotations` allows this on python 3.8+

**Pull request merged**